cluster/ubuntu/util.sh

```diff
@@ -450,7 +450,61 @@ function kube-down {
 
 # Update a kubernetes cluster with latest source
 function kube-push {
-  echo "not implemented"
+  #stop all the kube's process & etcd 
+  KUBE_ROOT=$(dirname "${BASH_SOURCE}")/../..
+  source "${KUBE_ROOT}/cluster/ubuntu/${KUBE_CONFIG_FILE-"config-default.sh"}"
+
+  for i in ${nodes}; do
+    {
+      echo "Cleaning on node ${i#*@}"
+      ssh -t $i 'sudo -p "[sudo] stop the all process: " service etcd stop'
+      ssh -t $i 'rm -f /opt/bin/kube* /etc/init/kube* /etc/init.d/kube* /etc/default/kube*; rm -rf ~/kube'
+    }
+  done
+  wait
+
+  # The latest version tag of kubernetes can be got by gsutil, or set in config-default.sh 
+  K8S_VERSION=$(gsutil cat gs://kubernetes-release/release/latest.txt)
+  echo "Using latest stable version: ${KUBE_VERSION}" >&2
+
+  # Run build.sh to get the latest release
+  source "${KUBE_ROOT}/cluster/ubuntu/build.sh"
+ 
+  #Update all nodes with the latest release
+  if [ ! -f "ubuntu/binaries/master/kube-apiserver" ]; then
+    echo "The latest release of kubernetes has not been got normally,plz check first"
+    exit 1
+  fi
+
+  setClusterInfo
+  ii=0
+
+  for i in ${nodes}
+  do
```

Comment:
Please put do on the same line as for:

https://google-styleguide.googlecode.com/svn/trunk/shell.xml?showone=Loops#Loops